lib/Sema/CSDiag.cpp

```diff
@@ -2273,6 +2273,32 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
       } else {
         // Otherwise the static member lookup was invalid because it was
         // called on an instance
+        
+        // Handle enum element lookup on instance type
+        auto lookThroughBaseObjTy = baseObjTy->lookThroughAllAnyOptionalTypes();
+        if (lookThroughBaseObjTy->is<EnumType>()
+            || lookThroughBaseObjTy->is<BoundGenericEnumType>()) {
+          for (auto cand : result.UnviableCandidates) {
+            ValueDecl *decl = cand.first;
+            if (auto enumElementDecl = dyn_cast<EnumElementDecl>(decl)) {
+              auto parentEnum = enumElementDecl->getParentEnum();
+              std::string enumTypeName;
+              llvm::raw_string_ostream typeNameStream(enumTypeName);
+              typeNameStream << parentEnum->getNameStr();
+              if (auto genericParams = parentEnum->getGenericParams()) {
+                genericParams->print(typeNameStream);
```

Comment:
Won't this include constraints and things we don't want? (Please add that to your test either way.)

Actually, in 99% of cases the generic parameters will be assumed anyway, so maybe it'd be better to just leave them out.